include/rspq.h

```diff
@@ -0,0 +1,704 @@
+ * moment a high-priority queue is created via #rspq_highpri_begin, the RSP
+ * immediately suspends execution of the command queue, and switches to
+ * the high-priority queue, waiting for commands. All commands added via
+ * standard APIs (#rspq_write) are then directed
+ * to the high-priority queue, until #rspq_highpri_end is called. Once the
+ * RSP has finished executing all the commands enqueue in the high-priority
+ * queue, it resumes execution of the standard queue.
+ * 
+ * The net effect is that commands enqueued in the high-priority queue are
+ * executed right away by the RSP, irrespective to whatever was currently
+ * enqueued in the standard queue. This can be useful for running tasks that
+ * require immediate execution, like for instance audio processing.
+ * 
+ * If required, it is possible to call #rspq_highpri_sync to wait for the
+ * high-priority queue to be fully executed.
+ * 
+ * Notice that the RSP cannot be fully preempted, so switching to the high-priority
+ * queue can only happen after a command has finished execution (before starting
+ * the following one). This can have an effect on latency if a single command
+ * has a very long execution time; RSP overlays should in general prefer
+ * providing smaller, faster commands.
+ * 
+ * This feature should normally not be used by end-users, but by libraries
+ * in which a very low latency of RSP execution is paramount to their workings.
+ * 
+ */
+
+#ifndef __LIBDRAGON_RSPQ_H
+#define __LIBDRAGON_RSPQ_H
+
+#include <stdint.h>
+#include <rsp.h>
+#include <pputils.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Maximum size of a command (in 32-bit words). */
+#define RSPQ_MAX_COMMAND_SIZE          16
+
+
+/**
+ * @brief A preconstructed block of commands
+ * 
+ * To improve performance of execution of sequences of commands, it is possible
+ * to create a "block". A block is a fixed set of commands that is created
+ * once and executed multiple times.
+ * 
+ * To create a block, use #rspq_block_begin and #rspq_block_end. After creation,
+ * you can use #rspq_block_run at any point to run it. If you do not need the
+ * block anymore, use #rspq_block_free to dispose it.
+ */
+typedef struct rspq_block_s rspq_block_t;
+
+/**
+ * @brief A syncpoint in the queue
+ * 
+ * A syncpoint can be thought of as a pointer to a position in the command queue.
+ * After creation, it is possible to later check whether the RSP has reached
+ * that position or not.
+ * 
+ * To create a syncpoint, use #rspq_syncpoint that returns a syncpoint that
+ * references the current position. Call #rspq_check_syncpoint or #rspq_wait_syncpoint
+ * to respectively do a single check or block waiting for the syncpoint to be
+ * reached by RSP.
+ * 
+ * Syncpoints are implemented using interrupts, so they have a light but non
+ * trivial overhead. Do not abuse them. For instance, it is reasonable to use
+ * tens of syncpoints per frame, but not hundreds or thousands of them.
```

Comment:
So far, syncpoints are not used internally anywhere. The mixer used to, but since it always pushes onto the high-priority queue, it uses highpri_sync instead. I feel like that is an implementation detail and doesn't really need to be documented anywhere.